libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -1266,18 +1270,30 @@ void CompilerUtils::rightShiftNumberOnStack(unsigned _bits)
 
 unsigned CompilerUtils::prepareMemoryStore(Type const& _type, bool _padToWords)
 {
+	solAssert(
+		_type.sizeOnStack() == 1,
+		"Memory store of types with stack size != 1 not implemented."
+	);
+
 	unsigned numBytes = _type.calldataEncodedSize(_padToWords);
+
+	solAssert(
+		numBytes > 0,
+		"Memory store of 0 bytes requested (Type: " + _type.richIdentifier() + ")."
```

Comment:
please use `toString(true)` instead.

Reply:
The reason is that `richIdentifier` create a string that is meant for machine consumption while `toString` targets humans.